src/Hazelcast.Net/Clustering/Cluster.cs

```diff
@@ -182,7 +182,14 @@ public async Task ConnectAsync(CancellationToken cancellationToken)
                 throw new ConnectionException("Failed to connected (aborted).");
 
             // connect
-            await Connections.ConnectAsync(cancellationToken).CfAwait();
+            try
+            {
+                await Connections.ConnectAsync(cancellationToken).CfAwait();
+            }
+            catch (ClientNotAllowedInClusterException e)
```

Comment:
That was because of trying to change the state  at `Cluster` level. I thought would be more organized since we play with the puppet at high level, like connecting the wires etc. Now,  I moved it to `ClusterConnections` where we set `Started` state. Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/587/commits/66b28e3e6d08210f07fe686b7eee389cfdcc3665